apps/desktop/src/key-management/biometrics/os-biometrics-mac.service.ts

```diff
@@ -21,26 +20,48 @@ export default class OsBiometricsServiceMac implements OsBiometricService {
     }
   }
 
-  async getBiometricKey(service: string, key: string): Promise<string | null> {
+  async isBiometricsAvailable(): Promise<boolean> {
+    return systemPreferences.canPromptTouchID();
+  }
+
+  async unlockWithBiometrics(userId: string, _clientkeyHalfB64: string): Promise<string | null> {
     const success = await this.authenticateBiometric();
 
     if (!success) {
-      throw new Error("Biometric authentication failed");
+      return null;
     }
 
-    return await passwords.getPassword(service, key);
+    return await passwords.getPassword(serviceName, userId + storageKeySuffix);
   }
 
-  async setBiometricKey(service: string, key: string, value: string): Promise<void> {
-    if (await this.valueUpToDate(service, key, value)) {
-      return;
+  async enableBiometricUnlock(
+    userId: string,
+    unlockKey: string,
+    _clientkeyHalfB64: string,
```

Comment:
🎨 typo and let's correct the types
```suggestion
  async unlockWithBiometrics(userId: string, _clientKeyHalfB64: string | undefined): Promise<string | null> {
```

Reply:
Just updated the suggestion, pasted the wrong one before.